pkg/apiserver/apiserver.go

```diff
@@ -168,6 +170,14 @@ func InstallLogsSupport(mux Mux) {
 	mux.Handle("/logs/", http.StripPrefix("/logs/", http.FileServer(http.Dir("/var/log/"))))
 }
 
+func InstallServiceErrorHandler(container *restful.Container) {
+	container.ServiceErrorHandler(serviceErrorHandler)
+}
+
+func serviceErrorHandler(serviceErr restful.ServiceError, response *restful.Response) {
+	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), latest.Codec, response)
```

Comment:
Note that we are using latest.Codec here, irrespective of the version for which the request was. (Thats because we do not have access to request here)
This can lead to situations like https://github.com/GoogleCloudPlatform/kubernetes/issues/6876, where users request a v1beta1 url and we will reply with v1beta3 status.